lasagne/tests/layers/test_special.py

```diff
@@ -446,3 +446,107 @@ def test_prelu(self, init_alpha):
+        out = layer.get_output_for(input, deterministic=True)
+        assert np.allclose(out, -0.5)
+
+    def test_dim_None(self, RandomizedRectifierLayer):
+        import lasagne
+        l_in = lasagne.layers.input.InputLayer((None, 3, 28, 28))
+        layer = RandomizedRectifierLayer(l_in)
+        input = np.ones((3, 3, 28, 28)).astype(theano.config.floatX)
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out, 1.0)
+
+    def assert_between(self, layer, input, output):
+        from numpy.testing import assert_approx_equal
+        x = input.flatten()
+        y = output.flatten()
+        for i in range(len(x)):
+            if x[i] >= 0:
+                assert_approx_equal(y[i], x[i])
+            else:
+                assert y[i] <= layer.lower * x[i] and y[i] > layer.upper * x[i]
+
+    def test_get_output_for(self, RandomizedRectifierLayer):
+        input_shape = (3, 3, 28, 28)
+
+        # ensure slope never exceeds [lower,upper)
+        input = np.random.randn(*input_shape).astype(theano.config.floatX)
+        layer = RandomizedRectifierLayer(input_shape, shared_axes=0)
+        self.assert_between(layer, input, layer.get_output_for(input).eval())
+
+        # from here on, we want to check parameter sharing
+        # this is easier to check if the input is all ones
+        input = np.ones(input_shape).astype(theano.config.floatX) * -1
+
+        # default: parameters shared along all but 2nd axis
+        layer = RandomizedRectifierLayer(input_shape)
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out[0].flatten(), out[-1].flatten())
+        assert not np.allclose(out[0][0].flatten(), out[0][-1].flatten())
+        assert np.allclose(out[0][0][0], out[0][0][-1])
+
+        # share across all axes (single slope)
+        layer = RandomizedRectifierLayer(input_shape, shared_axes='all')
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out[0].flatten(), out[-1].flatten())
+        assert np.allclose(out[0][0].flatten(), out[0][-1].flatten())
+        assert np.allclose(out[0][0][0], out[0][0][-1])
+
+        # share across 1st axis
+        layer = RandomizedRectifierLayer(input_shape, shared_axes=0)
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out[0].flatten(), out[-1].flatten())
+        assert not np.allclose(out[0][0].flatten(), out[0][-1].flatten())
+        assert not np.allclose(out[0][0][0], out[0][0][-1])
+
+        # share across 1st and 4th axes
+        layer = RandomizedRectifierLayer(input_shape, shared_axes=(0, 3))
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out[0].flatten(), out[-1].flatten())
+        assert not np.allclose(out[0][0].flatten(), out[0][-1].flatten())
+        assert not np.allclose(out[0][0][0].flatten(), out[0][0][-1].flatten())
+        assert np.allclose(out[0][0][0], out[-1][0][0])
+
+    def test_rrelu(self):
+        import lasagne
```

Comment:
Not sure it matters, but we usually just import what's needed in the tests:
```python
from lasagne.layers import InputLayer, DenseLayer, rrelu
from lasagne.nonlinearities import identity
```
(Haven't checked whether the ParametricRectifyLayer does that. It's probably not super important, but this way it clearly states which components are involved in a particular test.)